package.json

```diff
@@ -1,6 +1,6 @@
 {
   "name": "postmark-cli",
-  "version": "1.0.0",
+  "version": "2.0.0",
```

Comment:
I think this version bump is too high. First digit is reserved for major changes, we could probably go by something like 1.1.0 since there is a new feature being released.

Reply:
Ah, though I just noticed you introduced it due folder structure.

Reply:
Is there a way to do this without breaking existing integrations? Perhaps putting the "layouts" folder inside of the root download folder, or specifying the type as "Layout" within the metadata.json for the template? I think there are probably some other benefits to not splitting them into separate folders at the root.

Reply:
I agree with not breaking existing integrations. Also my concern here is already having version 2 for app that is 1 month old. The feature is not that big in terms of CLI to justify that big change. 

I do like though the folder split, since it follows the UI, and makes search through the list and separation easier. Although from what I see it would be easy to have them all in single folder since the name seems Alias based, therefore unique.

If we would like current new structure, there should be though an easy migration. All that would need to be done is check current folder structure (whether there is templates folder with subfolders) and provide a message to user that the files are migrated to new format.

Reply:
I agree that 2.0.0 is a huge bump for something so new, so I was reluctant about doing this.

We should still strive for separation between standard templates and layouts somehow to make layouts easy to find and avoid additional configuration in meta.json. I often find myself needing to quickly do spot fixes in the layout, so this would be helpful. I suggest that we go with Andrew's suggestion and just add a layouts folder to the list of standard templates in the root, so it's not a breaking change.

```
my-emails
├── _layouts
│   └── plain
│       ├── content.html
│       ├── content.txt
│       └── meta.json
└── password-reset
│   ├── content.html
│   ├── content.txt
│   └── meta.json
└── welcome
    ├── content.html
    └── meta.json
```

The reason for the underscore in the layouts folder name is so that it's more likely to appear at the top of the list. It's also to reduce the likelihood that it will conflict with existing aliases.

Thoughts?

Reply:
Thanks for being flexible, @derekrushforth. I do think that as a convention, placing files in these folders 

`_layouts`, `_partials`, etc. will be useful.

However, for a newcomer it may be non-obvious that the name of the folder is going to affect the type of template that gets saved. It also brings in another complication with moving templates, as changing the folder structure could cause the template type to change, which would be bad. If we think of each folder as being a "self contained" template, with all the info needed inside of it, maybe the problem gets easier...

By default, when we "pull" templates, this structure could work, but when "pushing" we're simply looking for directories that have the `meta.json`, `content.html`, and `content.txt` files. 

From there, meta.json handles any extra properties that need to be included, including `type`, but the `_layouts` directory is purely a default convention we provide, rather than a requirement to maintain template information.

It's certainly more complicated to resolve all of the templates when pushing/pulling, but this would give a lot of flexibility in the customer grouping these templates into sub-groups, etc on their end.

Reply:
@atheken definitely get what you're saying here. I like the idea of letting people change the folder structure to whatever they want, but yeah, this complicates things quite a bit.

My biggest concern is how we handle the process for pulling templates. We will need to figure out where each template is specifically located if they decided not to use our default folder structure.

I'm not so concerned about pushing, though. It's easier for me to rely on the template's metadata file instead of assuming template types based on the folder structure.

I'll give this a shot and let you know how it goes.

Reply:
I think the big take away for the moment is that we don't "move" templates to new locations compared to the 1.0 release and we should include the template type in `meta.json` -- `Standard` would be assumed when not set. 

Beyond that, defaulting to using a "_layouts" folder for new layouts is a good option.

We can layer on the "make your own folder structure later."